test/Test/Pos/Ssc/GodTossing/ComputeSharesSpec.hs

```diff
@@ -44,13 +60,184 @@ spec = describe "computeSharesDistr" $ do
+        let gFrac = fromMaybe (error "Distribution isn't found")
+                              (HM.lookup stId sd) `divRat` totalDistr
+        (rFrac, gFrac)
+
+    combineUnfair (er1, c1, c2) (er2, c3, c4) =
+        (max er1 er2, c1 + c3, c2 + c4)
+    computeUnfairness :: Rational -> Rational -> Unfairness
+    computeUnfairness real generated
+        -- Strictly say constant 0.5 depends on number of shares required
+        -- to reaval of commitment.
+        -- Bad case of the first type when
+        -- nodes can reveal commitment using shares
+        -- but real coin distribution says that nodes can't do it.
+        | real < 0.5 && generated > 0.5 = (generated - 0.5, 1, 0)
+        -- Bad case of the second type when
+        -- nodes can't reveal commitment using shares of honest nodes
+        -- but real coin distribution says that nodes can do it.
+        | real > 0.5 && generated < 0.5 = (0.5 - generated, 0, 1)
+        | otherwise = (0, 0, 0)
+
+isDistrReasonable :: Word16 -> RichmenStakes -> Either TossVerFailure SharesDistribution -> Bool
+isDistrReasonable _ _ (Left _) = False
+isDistrReasonable mx rs (Right sd) =
+    -- (1) Sum of distribution is less than some reasonable number.
+    isLimitedBy mx sd &&
+    -- (2) Distribution is fair (see explanation of @isDistrFair@)
+    isDistrFair rs sd
+
+isDistrReasonableMax :: RichmenStakes -> Either TossVerFailure SharesDistribution -> Bool
+isDistrReasonableMax = isDistrReasonable 300
+
+maxCoin :: Coin
+maxCoin = maxBound @Coin
+
+(</>) :: Coin -> Word64 -> Coin
+(</>) c d = mkCoin $ unsafeGetCoin c `div` d
+
+(<->) :: Coin -> Word64 -> Coin
+(<->) c w = unsafeSubCoin c (mkCoin w)
+
+richmenStakesFromCoins :: [Coin] -> RichmenStakes
+richmenStakesFromCoins coins = do
+    let stakeholders = map unsafeAddressHash [0 :: Integer ..]
+    HM.fromList $ zip stakeholders coins
+
+richmenStakesFromFractions :: [Double] -> RichmenStakes
+richmenStakesFromFractions fracts0 = do
+    let sm = sum fracts0
+    let ports = map (unsafeCoinPortionFromDouble . (/ sm)) fracts0
+    let coins = map (flip applyCoinPortionDown maxCoin) ports
+    richmenStakesFromCoins coins
+
+----------------------------------------------------------------------------
+-- Tests on total amount of shares
+----------------------------------------------------------------------------
+
+oneRichRichman :: Bool
+oneRichRichman = isDistrReasonable 1 richmen $ computeShares' richmen
+  where
+    richmen = richmenStakesFromCoins [maxCoin]
+
+twoRichRichmen :: Bool
+twoRichRichmen = isDistrReasonable 2 richmen $ computeShares' richmen
+  where
+    richmen = richmenStakesFromCoins [maxCoin </> 2 <-> 1, maxCoin </> 2]
+
+onePoorOneRich :: Bool
+onePoorOneRich = isDistrReasonable 100 richmen $ computeShares' richmen
+  where
+    richmen = richmenStakesFromFractions [0.01, 0.99]
+
+severalPoorOneRichRichmen :: Bool
+severalPoorOneRichRichmen = isDistrReasonable 100 richmen $ computeShares' richmen
```

Comment:
According to my computations in this case it's sufficient to generate 1, 5 and 13 shares. So 100 doesn't seem quite reasonable.